indexer/controller/coalesce.go

```diff
@@ -11,75 +11,79 @@ import (
 		}
-		// pack artifacts var
+		// Pack "artifacts" variable.
 		for _, layer := range s.manifest.Layers {
 			la := &indexer.LayerArtifacts{
 				Hash: layer.Hash,
 			}
 			var vscnrs indexer.VersionedScanners
 			vscnrs.PStoVS(pkgScanners)
-			// get packages from layer
-			pkgs, err := s.Store.PackagesByLayer(cctx, layer.Hash, vscnrs)
+			// Get packages from layer.
+			pkgs, err := s.Store.PackagesByLayer(gctx, layer.Hash, vscnrs)
 			if err != nil {
-				// on an early return cctx is canceled, and all inflight coalescers are canceled as well
+				// On an early return gctx is canceled, and all in-flight
+				// Coalescers are canceled as well.
 				return Terminal, fmt.Errorf("failed to retrieve packages for %v: %w", layer.Hash, err)
 			}
 			la.Pkgs = append(la.Pkgs, pkgs...)
-			// Get repos that have been created via the package scanners
-			pkgRepos, err := s.Store.RepositoriesByLayer(cctx, layer.Hash, vscnrs)
+			// Get repos that have been created via the package scanners.
+			pkgRepos, err := s.Store.RepositoriesByLayer(gctx, layer.Hash, vscnrs)
 			if err != nil {
 				return Terminal, fmt.Errorf("failed to retrieve repositories for %v: %w", layer.Hash, err)
 			}
 			la.Repos = append(la.Repos, pkgRepos...)
 
-			// get distributions from layer
-			vscnrs.DStoVS(distScanners) // method allocates new vscnr underlying array, clearing old contents
-			dists, err := s.Store.DistributionsByLayer(cctx, layer.Hash, vscnrs)
+			// Get distributions from layer.
+			vscnrs.DStoVS(distScanners) // Method allocates new "vscnr" underlying array, clearing old contents.
+			dists, err := s.Store.DistributionsByLayer(gctx, layer.Hash, vscnrs)
 			if err != nil {
 				return Terminal, fmt.Errorf("failed to retrieve distributions for %v: %w", layer.Hash, err)
 			}
 			la.Dist = append(la.Dist, dists...)
-			// get repositories from layer
+			// Get repositories from layer.
 			vscnrs.RStoVS(repoScanners)
-			repos, err := s.Store.RepositoriesByLayer(cctx, layer.Hash, vscnrs)
+			repos, err := s.Store.RepositoriesByLayer(gctx, layer.Hash, vscnrs)
 			if err != nil {
 				return Terminal, fmt.Errorf("failed to retrieve repositories for %v: %w", layer.Hash, err)
 			}
 			la.Repos = append(la.Repos, repos...)
-			// get files from layer
+			// Get files from layer.
 			vscnrs.FStoVS(fileScanners)
-			files, err := s.Store.FilesByLayer(cctx, layer.Hash, vscnrs)
+			files, err := s.Store.FilesByLayer(gctx, layer.Hash, vscnrs)
 			if err != nil {
 				return Terminal, fmt.Errorf("failed to retrieve files for %v: %w", layer.Hash, err)
 			}
 			la.Files = append(la.Files, files...)
-			// pack artifacts array in layer order
+			// Pack artifacts array in layer order.
 			artifacts = append(artifacts, la)
 		}
-		coalescer, err := ecosystem.Coalescer(cctx)
+		coalescer, err := ecosystem.Coalescer(gctx)
 		if err != nil {
 			return Terminal, fmt.Errorf("failed to get coalescer from ecosystem: %v", err)
 		}
-		// dispatch coalescer
+		// Dispatch.
```

Comment:
Not a part of this PR but I realized during this review that this pattern is pretty strange, we do all the expensive DB look ups for the coalescing context in serial and then do the (pretty cheap) coalescing in parallel, I suspect it's impact is negligible, what do you think?